policybot/mgrs/syncmgr/mgr.go

```diff
@@ -1277,24 +1277,32 @@ func (ss *syncState) handlePostSubmitTestResults() error {
 				FeatureLabels = append(FeatureLabels, featureLabel...)
 			}
 			fmt.Printf("saving PostSubmitTestResult batch of size %d\n", len(testResults))
-			err := ss.mgr.store.WritePostSumbitTestResults(ss.ctx, testResults)
+			err = ss.mgr.store.WritePostSumbitTestResults(ss.ctx, testResults)
 			if err != nil {
-				return fmt.Errorf("unable to write PostSumbitTestResults: %v", err)
+				return nil, fmt.Errorf("unable to write PostSumbitTestResults: %v", err)
 			}
 			err = ss.mgr.store.WriteSuiteOutcome(ss.ctx, SuiteOutcomes)
 			if err != nil {
-				return fmt.Errorf("unable to write SuiteOutcome: %v", err)
-			}
-			err = ss.mgr.store.WriteTestOutcome(ss.ctx, TestOutcomes)
-			if err != nil {
-				return fmt.Errorf("unable to write TestOutcome: %v", err)
+				return nil, fmt.Errorf("unable to write SuiteOutcome: %v", err)
 			}
-			err = ss.mgr.store.WriteFeatureLabel(ss.ctx, FeatureLabels)
-			if err != nil {
-				return fmt.Errorf("unable to write FeatureLabel: %v", err)
+			testOutcomeFeatureLabel := &storage.TestOutcomeFeatureLabel{}
+			testOutcomeFeatureLabel.TestOutcome = TestOutcomes
+			testOutcomeFeatureLabel.FeatureLabel = FeatureLabels
+			return testOutcomeFeatureLabel, nil
+		}).Batch(5).To(func(input interface{}) error {
+			for _, i := range input.([]interface{}) {
+				err := ss.mgr.store.WriteTestOutcome(ss.ctx, i.(*storage.TestOutcomeFeatureLabel).TestOutcome)
```

Comment:
This is still writing all testoutcomes for one suite in a single operation, and that number may be very high (1000+) for a single suite... have you run into any errors running this locally?

Reply:
I deleted all batches locally and change it to WithParallelism. It's still running and haven't got any error yet.

Reply:
I changed the batch of PostSubmitTestOutcomes to 5 and batch of TestOutcomes and feature lables to 1. It works, and my cloudtop still is running.

Reply:
Ok.  We may need to watch for future failures, and divide into smaller batches again, but this seems good enough to me.